File: game-server/src/main/java/org/forfun/mmorpg/game/cross/router/RandomBalanceStrategy.java
package org.forfun.mmorpg.game.cross.router;

import org.springframework.util.CollectionUtils;

import java.util.List;
import java.util.Random;

public class RandomBalanceStrategy implements BalanceStrategy {

    private Random random = new Random();

    @Override
    public int next(List<Integer> nodes) {
        if (CollectionUtils.isEmpty(nodes)) {
            return 0;
        }
        return nodes.get(random.nextInt() % nodes.size());
    }
}

File: game-server/src/main/kotlin/org/forfun/mmorpg/game/cross/router/RandomBalanceStrategy.kt
package org.forfun.mmorpg.game.cross.router

import org.springframework.util.CollectionUtils
import java.util.Random

class RandomBalanceStrategy : BalanceStrategy {

    private val random = Random()

    override fun next(nodes: List<Int>): Int {
        if (CollectionUtils.isEmpty(nodes)) {
            return 0
        }
        return nodes[random.nextInt() % nodes.size]
    }
}
